### ISSUE TITLE
How to save_weights of QAT model in tensorflow model optimization?

### ISSUE THREAD

**Issue by MATTYGILO** [tensorflow/tensorflow#59274]
<details><summary>Click to expand!</summary> 
 
 ### Issue Type

Support

### Have you reproduced the bug with TF nightly?

No

### Source

source

### Tensorflow Version

tf 2.10

### Custom Code

Yes

### OS Platform and Distribution

Linux Ubuntu 22.04

### Python version

3.9.13

### Current Behaviour?

```shell
I am trying to train a model using quantise awareness training. I am using tensorflow model optimization. I create a model, quantise it, fit it, save the weight.

Then at a later date I redefine and quantise and try to load the weights.

However, the model starts the whole training process from the start again.

This article explains how to save the whole QAT model.
https://www.tensorflow.org/model_optimization/guide/quantization/training_comprehensive_guide#checkpoint_and_deserialize

However, I wish to save the weights not model. Please don't suggest to follow the save whole model! All help would be much appreciated.
```

### Standalone code to reproduce the issue

```shell
from tensorflow_model_optimization.quantization.keras import quantise_model
model = define_model()
qat_model = quantize_model(model)
qat_model.fit(...)
qat_model.save_weights("qat_weights.h5")
... Finish for Now ...


... Pick up at a later date ...
model = define_model()
qat_model = quantize_model(model)
qat_model.load_weights("qat_weights.h5")
```

### Relevant log output

Starts the whole training process from the start at 0%

_No response_</details>

**Comment by pjpratik** [comment]
@MATTYGILO I tried to reproduce the issue using [Quantization aware training in Keras example](https://www.tensorflow.org/model_optimization/guide/quantization/training_example). 
 The `load_weights` seems to be working fine. The model trains from same train accuracy and val accuracy at the point of saving the model. Could you please check this [gist](https://colab.research.google.com/gist/pjpratik/9398ffd3c14ea7f60160353a221ddc86/59274.ipynb) and let us know if it helps. Thank you.

**Comment by MATTYGILO** [comment]
> @MATTYGILO! I tried to reproduce the issue using [Quantization aware training in Keras example](https://www.tensorflow.org/model_optimization/guide/quantization/training_example). The `load_weights` seems to be working fine.
> 
> The model trains from same train accuracy and val accuracy at the point of saving the model. Could you please check this [gist](https://colab.research.google.com/gist/pjpratik/9398ffd3c14ea7f60160353a221ddc86/59274.ipynb) and let us know if it helps.
> 
> Thank you.

I tried to reproduce the issue too with that notebook and all seems to be fine. Strange. I'll look specifically into my own case and get back to you.

**Comment by 1515236544hit** [comment]
Is this issue is fixed or any further proceeding is required?
Please let me know I am genuinely interested in solving this issue.

**Comment by pjpratik** [comment]
@MATTYGILO Could you please let us know if the issue still persists.

 Feel free to close the issue if the issue is resolved. Thanks!
